### PR TITLE
Display content information metrics 

### LINKED ISSUE #506: Display word, character count on overflow menu Action Sheet
On Aztec iOS, we recently added _word_ and _character_ counts as a description to the Action Sheet that is shown when pressing on the `•••` menu. I'd prefer to maintain this in the same location for Gutenberg if possible. For reference:

img_1647 trim 2019-01-25 22_57_23

Note: I'm not sure if/how we inherit this type of Aztec improvement in parallel, but I'd like to make sure we maintain them if possible for the sake of user expectations.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #506

 - WordPress-iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14305
 - WordPress-Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/12186
   - Follow up Android PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2383
   - Follow up WP-Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/12191
 - Gutenberg PR: https://github.com/WordPress/gutenberg/pull/23115

This PR changes the bridge method to display the content information metrics about the content being edited:

| iOS | Android |
|-----|-----------|
|  |  |

To test:
 - Start the demo app
 - iOS: Press the ... button on the top right, For Android you need to use the main app.
 - Check if the content metrics show on the top of the action sheet.

Notes:
 I considered implementing this in the main apps only, but in the end, I thought it was better to use a single source of the metrics for all three implementations: web, iOS, Android. 
If we implemented in the main apps we will need to replicate the code in three places, and the only marginal advantage could be some performance, wich I don't think it's worth the maintenance costs.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by iamthomasbishop** [comment]
Based on some discussion on Slack, I whipped up some options to discuss. For reference, here are the items that the web editor provides:

<a href="https://user-images.githubusercontent.com/1200432/84516581-9cdb2900-ac93-11ea-83e4-277e673575c4.png"></a>

* Words
* Headings
* Paragraph
* Blocks

We have character and word count on Aztec (the "classic" editor) but not on mobile Gutenberg, so we are looking at some or all of the following:

* Words
* Characters
* Headings
* Paragraphs
* Blocks

Here are the options, based on the above:

<a href="https://user-images.githubusercontent.com/1200432/84517142-723da000-ac94-11ea-98f5-702b27ad6367.png"></a>

* Current: no content info
* Option 1: blocks, words, characters
* Option 2: blocks, paragraphs, words, characters
* Option 3: blocks, headings, paragraphs, words, characters

To keep things simple and build upon what already exists for users on Aztec, I think the best approach is to start with Option 1 (blocks, words, characters). Option 3 makes sense if we want to fully align with core and Option 2 feels stuck in-between.

// cc @SergioEstevao @hypest

**Comment by SergioEstevao** [comment]
@iamthomasbishop I implemented option 1 above and just updated the screenshot of the description.

**Comment by iamthomasbishop** [comment]
@SergioEstevao Is it possible to only add the plural ("s") when the given type's value > 1? In other words "1 block" vs. "2 blocks", etc.

**Comment by hypest** [comment]
👋 @SergioEstevao , I've opened this [gb-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2383) and this [WPAndroid PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12191) (targeting your PRs) to complete the Android side support of the feature. Feel free to merge to your PRs before or after they get reviewed.

Notice, I've targeted the 15.2 WP apps milestone as I don't think the feature can be merged in time for the 15.1. You might want to adjust the milestone and the related RELEASE-NOTES.txt mention on your PRs too.

**Comment by SergioEstevao** [comment]
> @SergioEstevao Is it possible to only add the plural ("s") when the given type's value > 1? In other words "1 block" vs. "2 blocks", etc.

@iamthomasbishop so at the moment in Aztec we everything in the plural always: Ex:` 1 words, 1 character`.

This is because of a problem with WordPress translating tools (Glotpress) that don't support plurality as you can read here: https://github.com/wordpress-mobile/WordPress-iOS/issues/6327

I was trying to go around the issue by introducing the `(s)` what do you think it's the best way to proceed?

File: react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift [hunk 1]
```diff
@@ -8,9 +8,16 @@ public class RNReactNativeGutenbergBridge: RCTEventEmitter {
...
+    func provideToNative_Html(_ html: String, title: String, changed: Bool, contentInfo: [String:Int]) {
         DispatchQueue.main.async {
-            self.delegate?.gutenbergDidProvideHTML(title: title, html: html, changed: changed)
+            var info: ContentInfo?
+            if  let characters = contentInfo["characterCount"],
+                let words = contentInfo["wordCount"],
+                let paragraphs = contentInfo["paragraphCount"],
+                let blocks = contentInfo["blockCount"] {
+                info = ContentInfo(characterCount: characters, wordCount: words, paragraphCount: paragraphs, blockCount: blocks)
+            }
```
Comment (etoledom) [inline react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift:19]: Maybe we can do this as a `ContentInfo` struct extension?
In that way we clear up a bit this code.

Maybe we could also `Decodable` but just if it's not more complex (or an overkill) 🤔
What I like from Decodable is that for any change in the Object from JS, we just need to do one change in one place (the definition struct itself).

Comment (SergioEstevao) [reply react-native-gutenberg-bridge/ios/RNReactNativeGutenbergBridge.swift:19]: Good suggestion!

File: react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java [hunk 2]
```diff
@@ -730,6 +739,38 @@ public CharSequence getTitle(OnGetContentTimeout onGetContentTimeout) {
...
+            }
+
+            mGetContentCountDownLatch = new CountDownLatch(1);
+
+            mRnReactNativeGutenbergBridgePackage.getRNReactNativeGutenbergBridgeModule().getHtmlFromJS();
+
+            new Thread(new Runnable() {
+                @Override public void run() {
+                    try {
+                        mGetContentCountDownLatch.await(5, TimeUnit.SECONDS);
```
[the remark this answers is not in the document]
Comment (hypest) [reply react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java:756]: Indeed, I opted for a lower countdown as this is related to a UI interaction, and I thought that the shorter the timeout the better so user has more timely feedback on what happens/happened with the attempt to show the Content Info.

I chose 5secs rather arbitrarily at that though. Not too fast and not too long to start to feel weird on mobile. What do you think?

[the remark this answers is not in the document]
Comment (mchowning) [reply react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java:756]: Makes sense. 👍  Thanks for explaining.

**Pull request merged**